src/taipy/config/_config_comparator/_config_comparator.py

```diff
@@ -0,0 +1,156 @@
+# Copyright 2023 Avaiga Private Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+# the License. You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+import json
+from copy import copy
+from typing import Optional, Set, Union
+
+from deepdiff import DeepDiff
+
+from ...logger._taipy_logger import _TaipyLogger
+from .._config import _Config
+from .._serializer._json_serializer import _JsonSerializer
+from ..exceptions.exceptions import ConflictedConfigurationError
+from ._comparator_result import _ComparatorResult
+
+
+class _ConfigComparator:
+
+    _UNCONFLICTED_SECTIONS: Set[str] = set()
+
+    __logger = _TaipyLogger._get_logger()
+
+    @classmethod
+    def _add_unconflicted_section(cls, section_name: Union[str, Set[str]]):
+        if isinstance(section_name, str):
+            section_name = {section_name}
+
+        cls._UNCONFLICTED_SECTIONS.update(section_name)
+
+    @classmethod
+    def _compare(
+        cls,
+        old_config: _Config,
+        new_config: _Config,
+        old_version_number: Optional[str] = None,
+        new_version_number: Optional[str] = None,
+        raise_error: bool = True,
```

Comment:
If the client knows if an error should be raised, the client should take responsibility for raising the error.